### PR TITLE
SSL mode value set based in use_azure flag

### PR DESCRIPTION
**Pull request by VenkataChalla554**
## Context
Removed hard code value of SSL Mode.

## Changes proposed in this pull request
SSL Mode Value set based on user_azure. 

## Guidance to review
Deployed review app using this branch  : https://github.com/DFE-Digital/teaching-record-system/tree/1813-standardise-logging-terraform-module-test.

make dv_review terraform-apply  CLUSTER=cluster2 IMAGE=ghcr.io/dfe-digital/teaching-record-system:b96c149d3b2db5f4ce4ddb29928cfec6af0e286a  APP_NAME=2222 

 

## Checklist

- [x] I have performed a self-review of my code, including formatting and typos
- [x] I have [cleaned the commit history](https://www.annashipman.co.uk/jfdi/good-pull-requests.html)
- [x] I have added the `Devops` label
- [x] I have attached the pull request to the trello card

### CODE REVIEW

**Comment by VenkataChalla554** [comment]
[Standardise logging fields](https://trello.com/c/AqSayzVB/1813-standardise-logging-fields)

File: aks/postgres/outputs.tf [hunk 1]
```diff
@@ -26,15 +26,16 @@ output "name" {
...
 
 output "url" {
-  value     = "postgres://${urlencode(local.database_username)}:${urlencode(local.database_password)}@${local.host}:${local.port}/${local.database_name}"
+  value     = "postgres://${urlencode(local.database_username)}:${urlencode(local.database_password)}@${local.host}:${local.port}/${local.database_name}?sslmode=${var.use_azure ? "require" : "disable"}"
   sensitive = true
 }
 
 output "dotnet_connection_string" {
-  value     = "Server=${local.host};Database=${local.database_name};Port=${local.port};User Id=${local.database_username};Password='${local.database_password}';Ssl Mode=Require;Trust Server Certificate=true"
+  value     = "Server=${local.host};Database=${local.database_name};Port=${local.port};User Id=${local.database_username};Password='${local.database_password}';Ssl Mode=${var.use_azure ? "require" : "disable"};Trust Server Certificate=true"
```
Comment (saliceti) [inline aks/postgres/outputs.tf:34]: According to https://www.npgsql.org/doc/security.html?tabs=tabid-1 it should be Disable. Or Prefer, to always try SSL first.

File: aks/postgres/outputs.tf [hunk 2]
```diff
@@ -26,15 +26,16 @@ output "name" {
 }
 
 output "url" {
-  value     = "postgres://${urlencode(local.database_username)}:${urlencode(local.database_password)}@${local.host}:${local.port}/${local.database_name}"
+  value     = "postgres://${urlencode(local.database_username)}:${urlencode(local.database_password)}@${local.host}:${local.port}/${local.database_name}?sslmode=${var.use_azure ? "require" : "disable"}"
```
Comment (saliceti) [inline aks/postgres/outputs.tf:29]: Can we use prefer instead of disable, to always try SSL first

File: aks/postgres/outputs.tf (hunk 1 above)
Comment (VenkataChalla554) [reply aks/postgres/outputs.tf]: fixed

File: aks/postgres/outputs.tf (hunk 2 above)
Comment (VenkataChalla554) [reply aks/postgres/outputs.tf]: fixed

**Review by saliceti — APPROVED** [review]

**Pull request merged**